script.module.t1mlib/lib/t1mlib.py

```diff
@@ -126,9 +128,53 @@ def getAddonLiveVideo(self, url):
     def getAddonVideo(self, url):
         xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, xbmcgui.ListItem(path=uqp(url), offscreen=True))
 
+
     def doFunction(self, url):
         return
 
+
+    def cleanFilename(self, filename):
+        whitelist = "-_.() %s%s" % (string.ascii_letters, string.digits)
+        filename = ''.join(c for c in filename if c in whitelist)
+        return filename
+
+
+    def addMovieToLibrary(self, url):
+        name  = self.cleanFilename(xbmc.getInfoLabel('ListItem.Title'))
+        profile = self.addon.getAddonInfo('profile')
+        moviesDir  = xbmc.translatePath(os.path.join(profile,'Movies'))
+        movieDir  = xbmc.translatePath(os.path.join(moviesDir, name))
+        if not os.path.isdir(movieDir):
+            os.makedirs(movieDir)
+        strmFile = xbmc.translatePath(os.path.join(movieDir, name+'.strm'))
+        url = ''.join([sys.argv[0],'?mode=GV&url=',url])
+        with open(strmFile, 'w') as outfile:
+            outfile.write(url)
+        json_cmd = '{"jsonrpc":"2.0","method":"VideoLibrary.Scan", "params": {"directory":"%s/"},"id":1}' % movieDir.replace('\\','/')
```

Comment:
maybe `xbmcvfs.validatePath` instead of the movieDir.replace?

Reply:
Again what I'm doing here works across platforms.